docs/the-journey-of-a-cohort-item.md

```diff
@@ -0,0 +1,165 @@
+
+```
+CohortItem(
+    subscriptionName  = "S-00000003"
+    processingStage   = "SalesforcePriceRiceCreationComplete"
+    currency          = Some("EUR")
+    billingPeriod     = Some("Monthly")
+    oldPrice          = Some(BigDecimal(52))
+    estimatedNewPrice = Some(BigDecimal(61))
+    startDate         = Some(LocalDate.of(2024, 5, 10))
+)
+```
+
+Then, the cohort item is going to.... sleep. It's going to sleep as long as it take for it to be at the start date minus about 40 days. This might take a few days or up to a year.
+
+### Notification and Amendment
+
+Today is now `LocalDate.of(2024, 5, 10)` minus about 40 days. The engine sees a subscription in `SalesforcePriceRiceCreationComplete` stage ready to be user notified. The engine is going to perform a certain number of look ups and will send a message to a queue that will eventually be delivered to Braze (triggering the sending of an email, or sending an additional request to an external company for a letter to be printed and delivered).
+
+The message to the user is going to mention the start date and the new estimated new price. The engine then notifies Salesforce that the user has been notified (for record keeping) and then will perform the amendment in Zuora, meaning will update Zuora with the fact that the subscription in Zuora is price risen and that the price rise is taking effect on the date that had already been decided during the estimation step. Note that this is the first and only moment that the engine performs and write operation in Zuora during the entire price rise process of that subscription. 
+
+It is also important to notice that the amendment step only happened after the user notification step. This is to avoid a situation where there would be a bug in the engine or even just a long outage and causing subscriptions to be price risen in Zuora without the users having (yet) been notified. That would be illegal and put the Guardian in hot water.
+
+At this point the processing stage is `AmendmentComplete`.
+
+The last step is now another Salesforce update, where we inform Salesforce that the subscription in Zuora has been edited for price rise. Then the processing stage becomes `AmendmentWrittenToSalesforce`. This completes the price rise of subscription "S-00000003".
+
+### Cancellations
+
+As we have seen a cohort item / subscription can sleep for a long time before it is ready to move to notification process, but what happens if the subscription has been cancelled by the user in the meantime ?
+
+In such a case, the engine will detect that the subscription has been cancelled in Zuora and will move the cohort item to `Cancelled` processing stage
+
+```
+CohortItem(
+    subscriptionName  = "S-00000003"
+    processingStage   = "Cancelled"
+)
+```
+
+Once the cohort item is in `Cancelled` state the engine will no longer touch it. Alike `AmendmentWrittenToSalesforce`, `Cancelled` is a final state for a cohort item.
```

Comment:
Is there a state-transition diagram somewhere that shows what's possible and what the triggers are?